### PR TITLE
Detect package versions using importlib in napari information

### PR DESCRIPTION
**Pull request by Czaki**
# Description

In this PR, change, use `importlib.metadata.version` instead of module level `__version__` variable. 

The motivation for this PR is a problem with the version reporting of the latest superqt in #5417

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Review by Carreau — APPROVED** [review]
No objection, though just to note in my experience `importlib.metadata ` is often wrong when testing dev-versions/editable installs.

**Comment by Czaki** [comment]
After rethink, I just decided to improve the version detection mechanism to avoid using bare exception catch. It is an improvement, but I'm not sure if important enough.

File: napari/utils/info.py [hunk 1]
```diff
@@ -122,8 +122,13 @@ def sys_info(as_html=False):
...
-            text += f"<b>{name}</b>: {version(module)}<br>"
-        except PackageNotFoundError as e:
+            try:
+                loaded[module] = __import__(module)
+                text += f"<b>{name}</b>: {loaded[module].__version__}<br>"
+            except AttributeError:
+                text += f"<b>{name}</b>: {version(module)}<br>"
+
+        except (PackageNotFoundError, ImportError) as e:
             text += f"<b>{name}</b>: version detect failed failed ({e})<br>"
```
Comment (brisvag) [inline napari/utils/info.py]:
```suggestion
            text += f"<b>{name}</b>: version detect failed ({e})<br>"
```

**Review by brisvag — APPROVED** [review]
Would it be possible to test the failure case? Codecov marks as not covered, and maybe it good to have this covered.

But non-blocking, feel free to merge as is.

**Comment by GenevieveBuckley** [comment]
Are there any further changes planned here, or should this be merged now?

**Comment by Czaki** [comment]
There was a moment when this was required, but `__vesion__` in superqt was fixed. Maybe I should delete this PR.

**Comment by GenevieveBuckley** [comment]
> There was a moment when this was required, but `__vesion__` in superqt was fixed. Maybe I should delete this PR.

Ok, if the change is no longer needed then that sounds like a good plan.

**Pull request closed without merging**